src/ui/win/win_ui.cpp

```diff
@@ -0,0 +1,257 @@
+#include "win_ui.h"
+
+#ifdef _WIN32 // Windows
+
+#include <iostream>
+#include <string>
+
+namespace WindowsUi
+{
+    // This is a wrapper class which was made to avoid UiWindowWin class member exposure to public
+    class UiWindowWinInfo
+    {
+    public:
+        explicit UiWindowWinInfo( const UiWindowWin * window )
+            : _window( window )
+        {
+        }
+
+        const PenguinV_Image::Image & image() const
+        {
+            return _window->_image;
+        }
+
+        const BITMAPINFO * bitmapInfo() const
+        {
+            return _window->_bmpInfo;
+        }
+
+        bool valid() const
+        {
+            return (_window != nullptr) && !_window->_image.empty() && (_window->_bmpInfo != nullptr);
+        }
+
+        void paint( HDC hdc, RECT clientRoi ) const
+        {
+            const double xFactor = static_cast<double>(clientRoi.right - clientRoi.left) / _window->_image.width();
+            const double yFactor = static_cast<double>(clientRoi.bottom - clientRoi.top) / _window->_image.height();
+
+            for ( std::vector < UiWindowWin::PointToDraw >::const_iterator point = _window->_point.cbegin(); point != _window->_point.cend(); ++point ) {
```

Comment:
`for (const auto& point : _points)`

Reply:
There are big debates regarding the usage of `auto` word. Some people say that it';s shorter, another say that it makes code complicated as you don't know with which object you work with.

Reply:
Haha :). A big discussion in the `C++` community. For me it make sense in vector iteration as it's shorter and cleaner. I would never write` auto i = 5`;  instead of `int i = 5`;

Reply:
I can say from my own personal experience as I met multiple opinions from multiple companies. Let's stick to non c++11 code for now. No harm in this :)